hardware/Drivetrain.java

```diff
@@ -97,29 +86,25 @@ public void init(HardwareMap hwMap) {
     }
 
     private void setPower(double power) {
-        setPower(power, power, power, power);
+        setPower(new double[]{power, power, power, power});
     }
 
-    public void setPower(double FL, double FR, double BL, double BR) {
-        frontLeft.setPower(FL);
-        backRight.setPower(BR);
-        backLeft.setPower(BL);
-        frontRight.setPower(FR);
+    public void setPower(double[] powers) {
+        // sequence is frontLeft, frontRight, backLeft, backRight
+        DcMotor[] dcMotors = motors.values().toArray(new DcMotor[motors.size()]);
+        for(int motor = 0; motor < 4; dcMotors[motor].setPower(powers[motor]), motor++) { }
     }
 
     public void setMode(DcMotor.RunMode mode) {
-        frontLeft.setMode(mode);
-        backRight.setMode(mode);
-        backLeft.setMode(mode);
-        frontRight.setMode(mode);
+        for(DcMotor motor: motors.values()) { motor.setMode(mode); }
     }
 
     public void teleDrive(double r, double robotAngle, double rightX) {
-        double v1 = r * Math.sin(robotAngle) - rightX;
-        double v2 = r * Math.cos(robotAngle) + rightX;
-        double v3 = r * Math.cos(robotAngle) - rightX;
-        double v4 = r * Math.sin(robotAngle) + rightX;
-        setPower(v1,v2,v3,v4);
+        double v1 = r * Math.sin(robotAngle) - rightX; //frontLeft
+        double v2 = r * Math.cos(robotAngle) + rightX; //frontRight
+        double v3 = r * Math.cos(robotAngle) - rightX; //backLeft
+        double v4 = r * Math.sin(robotAngle) + rightX; //backRight
+        setPower(new double[]{v1, v2, v3, v4});
```

Comment:
refer to the demo branch to see if these values need to be negated or not. or see if the reverse driving / turning are affected on new drivetrain chassis.